Web/react-spa/src/views/MakeOrderView.tsx

```diff
@@ -0,0 +1,69 @@
+import authService from "app/authService"
+import { Toaster, toast } from 'react-hot-toast'
+import { FormEvent, useState } from 'react'
+
+export function MakeOrderView() {
+
+
+    const [state, setState] = useState<{ [index: string]: string }>({
+        street: '',
+        city: '',
+        state: '',
+        country: '',
+        zipCode: ''
+    })
+
+    const makeOrder = async (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+        const token = await authService.getToken()
+        const id = await authService.getUserId()
+        const url = process.env.REACT_APP_API_URL + `/api/agg/order/create/${id}`
+        const responce = await fetch(url, {
+            method: 'POST',
+            headers: {
+                'Authorization': `Bearer ${token}`,
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                street: state.street,
+                city: state.city,
+                state: state.state,
+                country: state.country,
+                zipCode: state.zipCode
+            })
+        })
+        if (responce.ok) {
+            toast.success('Order was made')
+        } else {
+            toast.error(await responce.text())
+        }
+    }
+
+    const handleChange = (event: FormEvent<HTMLInputElement>) => {
+        const newState = { ...state }
+        const key = event.currentTarget.name
+        newState[key] = event.currentTarget.value
+        setState(newState)
+    }
+
+    return (<section className="small-container">
+        <form onSubmit={makeOrder}>
+            <label htmlFor='country'>Country</label>
+            <input required onChange={handleChange} type='text' placeholder='country' name="country" id='country'></input>
```

Comment:
see, this repeats many times, so this could be a component. Like, for example InputGroup